### PR TITLE
ZCS-13605 : Upgrade ClamAV to 1.0.1

### PR DESCRIPTION
**Pull request by umagmrit**
This PR has been created on the top of OpenSSL-3.0.9 Upgrade PR (https://github.com/Zimbra/packages/pull/189) because Clamav needs to compile with OpenSSL-3.0.9.

- Upgrade ClamAV to 1.0.1
- Updated clamav-db for ClamAV-1.0.1

### CODE REVIEW

File: thirdparty/clamav-db/zimbra-clamav-db/debian/zimbra-clamav-db.postinst [hunk 1]
```diff
@@ -1,6 +1,15 @@
 #!/bin/bash
 if [ "$1" = configure ]; then
-  /bin/chown -R zimbra:zimbra /opt/zimbra/data/clamav
-  /bin/chmod 644 /opt/zimbra/data/clamav/init/*.init
+	/bin/chown -R zimbra:zimbra /opt/zimbra/data/clamav
+	/bin/chmod 644 /opt/zimbra/data/clamav/init/*.init
+	if [ -f /opt/zimbra/data/clamav/db/daily.cvd ]; then
```
Comment (silentsakky) [inline thirdparty/clamav-db/zimbra-clamav-db/debian/zimbra-clamav-db.postinst:5]: why are we removing these files?

File: thirdparty/clamav-db/Makefile [hunk 2]
```diff
@@ -11,9 +12,9 @@ mcvd := main.cvd
 bcvd_file := $(SRC_DIR)/$(bcvd)
 dcvd_file := $(SRC_DIR)/$(dcvd)
 mcvd_file := $(SRC_DIR)/$(mcvd)
-burl := http://database.clamav.net/$(bcvd)
-durl := http://database.clamav.net/$(dcvd)
-murl := http://database.clamav.net/$(mcvd)
+burl := https://files.zimbra.com/downloads/clamav-database/$(pvers)/$(bcvd)
```
Comment (silentsakky) [inline thirdparty/clamav-db/Makefile:15]: don't we have a public url from clamav for this?

File: thirdparty/clamav-db/zimbra-clamav-db/debian/zimbra-clamav-db.postinst [hunk 3]
```diff
@@ -1,6 +1,15 @@
...
+		rm -rf /opt/zimbra/data/clamav/db/daily.cvd
+	fi
+	if [ -f /opt/zimbra/data/clamav/db/main.cvd ]; then
```
Comment (silentsakky) [inline thirdparty/clamav-db/zimbra-clamav-db/debian/zimbra-clamav-db.postinst:8]: we are already doing this as part of package installation, do we need to repeat here?

File: thirdparty/clamav-db/Makefile (hunk 2 above)
Comment (umagmrit) [reply thirdparty/clamav-db/Makefile:15]: we have but clamav public url not allowed to directly download using wget and curl.
https://github.com/Cisco-Talos/clamav/issues/585#issuecomment-1127863490

File: thirdparty/clamav-db/zimbra-clamav-db/debian/zimbra-clamav-db.postinst (hunk 1 above)
Comment (umagmrit) [reply thirdparty/clamav-db/zimbra-clamav-db/debian/zimbra-clamav-db.postinst:5]: We need to remove these to copy updated db files.
https://github.com/Zimbra/zm-core-utils/blob/develop/src/bin/zmclamdctl#L80

File: thirdparty/clamav-db/zimbra-clamav-db/debian/zimbra-clamav-db.postinst (hunk 3 above)
Comment (umagmrit) [reply thirdparty/clamav-db/zimbra-clamav-db/debian/zimbra-clamav-db.postinst:8]: package installation won't install  *.cvd files. It installs *.cvd.init files. zmclamdctl script will copy the files(.cvd.init --> .cvd) if they are not present.

**Review by silentsakky — APPROVED** [review]

**Review by dasiyogesh — APPROVED** [review]

**Comment by stondino00** [comment]
Please upgrade clamav to 1.0.3 ASAP.  1.0.2 and 1.0.3 fix critical vulnerabilities.

1.0.3
ClamAV 1.0.3 is a critical patch release with the following fixes:

Upgrade the bundled UnRAR library (libclamunrar) to version 6.2.10.
GitHub pull request: https://github.com/Cisco-Talos/clamav/pull/1010


1.0.2 
ClamAV 1.0.2 is a critical patch release with the following fixes: 



[CVE-2023-20197](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2023-20197) Fixed a possible denial of service vulnerability in the HFS+ file parser. This issue affects versions 1.1.0, 1.0.1 through 1.0.0, 0.105.2 through 0.105.0, 0.104.4 through 0.104.0, and 0.103.8 through 0.103.0. Thank you to Steve Smith for reporting this issue. 
[CVE-2023-20212](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2023-20212) Fixed a possible denial of service vulnerability in the AutoIt file parser. This issue affects versions 1.0.1 and 1.0.0. This issue does not affect version 1.1.0. 
Fixed a build issue when using the Rust nightly toolchain, which was affecting the oss-fuzz build environment used for regression tests. 
GitHub pull request: https://github.com/Cisco-Talos/clamav/pull/996 
Fixed a build issue on Windows when using Rust version 1.70 or newer.
GitHub pull request: https://github.com/Cisco-Talos/clamav/pull/993 
CMake build system improvement to support compiling with OpenSSL 3.x on macOS with the Xcode toolchain. The official ClamAV installers and packages are now built with OpenSSL 3.1.1 or newer. 
GitHub pull request: https://github.com/Cisco-Talos/clamav/pull/973 
Fixed an issue where ClamAV does not abort the signature load process after partially loading an invalid signature. The bug would later cause a crash when scanning certain files. 
GitHub pull request: https://github.com/Cisco-Talos/clamav/pull/952 
Fixed an issue so that ClamAV correctly removes temporary files generated by the VBA and XLM extraction modules so that the files are not leaked in patched versions of ClamAV where temporary files are written directly to the temp-directory instead of writing to a unique subdirectory.
GitHub pull request: https://github.com/Cisco-Talos/clamav/pull/900 
Set Git attributes to prevent Git from altering line endings for bundled Rust libraries. Third-party Rust libraries are bundled in the ClamAV release tarball. We do not commit them to our own Git repository, but community package maintainers may now store the tarball contents in Git. The Rust build system verifies the library manifest, and this change ensures that the hashes are correct. Improvement courtesy of Nicolas R. 
GitHub pull request: https://github.com/Cisco-Talos/clamav/pull/856 
Fixed two bugs that would cause Freshclam to fail update when applying a CDIFF database patch if that patch adds a file to the database archive or removes a file from the database archive. This bug also caused Sigtool to fail to create such a patch. 
GitHub pull request: https://github.com/Cisco-Talos/clamav/pull/901

**Comment by stondino00** [comment]
Please look at upgrading bundled clamav.  Security fixes below.


ClamAV 1.2.2 and ClamAV 1.0.5 are critical patch versions with the following fixes:

[CVE-2024-20290](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2024-20290): Fixed a possible heap overflow read bug in the OLE2 file parser that could cause a denial-of-service (DoS) condition.

Thank you to OSS-Fuzz for identifying this issue.

Affected versions:
1.0.0 through 1.0.4 (LTS)
1.1 (all patch versions)
1.2.0 and 1.2.1
[CVE-2024-20328](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2024-20328): Fixed a possible command injection vulnerability in the "VirusEvent" feature of ClamAV's ClamD service.

To fix this issue, we disabled the '%f' format string parameter.  ClamD administrators may continue to use the `CLAM_VIRUSEVENT_FILENAME`  environment variable, instead of '%f'. But you should do so only from within  an executable, such as a Python script, and not directly in the clamd.conf "VirusEvent" command.

Thank you to Amit Schendel for identifying this issue.

Affected versions:
0.104 (all patch versions)
0.105 (all patch versions)
1.0.0 through 1.0.4 (LTS)
1.1 (all patch versions)
1.2.0 and 1.2.1

**Comment by stondino00** [comment]
Clamav has been updated to fix vulnerabilities.


[ClamAV 1.3.1, 1.2.3, 1.0.6 patch versions published](https://blog.clamav.net/2024/04/clamav-131-123-106-patch-versions.html)
Today, we are publishing the 1.3.1, 1.2.3, and 1.0.6 security patch versions. 

The release files for the patch versions are available for download on the [ClamAV downloads page](https://www.clamav.net/downloads), on the [GitHub Release page](https://github.com/Cisco-Talos/clamav/releases), and through [Docker Hub](https://hub.docker.com/r/clamav/clamav/).

The images on Docker Hub may not be immediately available on release day.

Continue reading to learn what changed in each version.

1.3.1
ClamAV 1.3.1 is a critical patch release with the following fixes:

[CVE-2024-20380](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2024-20380): Fixed a possible crash in the HTML file parser that could cause a denial-of-service (DoS) condition.

This issue affects version 1.3.0 only and does not affect prior versions.

Thank you to Błażej Pawłowski for identifying this issue.

[GitHub pull request](https://github.com/Cisco-Talos/clamav/pull/1242)
Updated select Rust dependencies to the latest versions. This resolved Cargo audit complaints and included PNG parser bug fixes.

[GitHub pull request](https://github.com/Cisco-Talos/clamav/pull/1227)
Fixed a bug causing some text to be truncated when converting from UTF-16.

[GitHub pull request](https://github.com/Cisco-Talos/clamav/pull/1230)
Fixed assorted complaints identified by Coverity static analysis.

[GitHub pull request](https://github.com/Cisco-Talos/clamav/pull/1235)
Fixed a bug causing CVDs downloaded by the DatabaseCustomURL Freshclam config option to be pruned and then re-downloaded with every update.

[GitHub pull request](https://github.com/Cisco-Talos/clamav/pull/1238)
Added the new 'valhalla' database name to the list of optional databases in preparation for future work.

[GitHub pull request](https://github.com/Cisco-Talos/clamav/pull/1238)
Added symbols to the libclamav.map file to enable additional build configurations.

Patch courtesy of Neil Wilson.

[GitHub pull request](https://github.com/Cisco-Talos/clamav/pull/1244)

1.2.3
ClamAV 1.2.3 is a critical patch release with the following fixes:

Updated select Rust dependencies to the latest versions. This resolved Cargo audit complaints and included PNG parser bug fixes.

[GitHub pull request](https://github.com/Cisco-Talos/clamav/pull/1226)
Fixed a bug causing some text to be truncated when converting from UTF-16.

[GitHub pull request](https://github.com/Cisco-Talos/clamav/pull/1231)
Fixed assorted complaints identified by Coverity static analysis.

[GitHub pull request](https://github.com/Cisco-Talos/clamav/pull/1236)
Fixed a bug causing CVDs downloaded by the DatabaseCustomURL Freshclam config option to be pruned and then re-downloaded with every update.

[GitHub pull request](https://github.com/Cisco-Talos/clamav/pull/1239)
Added the new 'valhalla' database name to the list of optional databases in preparation for future work.

[GitHub pull request](https://github.com/Cisco-Talos/clamav/pull/1239)
Silenced a warning "Unexpected early end-of-file" that occured when scanning some PNG files.

[GitHub pull request](https://github.com/Cisco-Talos/clamav/pull/1215)

1.0.6
ClamAV 1.0.6 is a critical patch release with the following fixes:

Updated select Rust dependencies to the latest versions. This resolved Cargo audit complaints and included PNG parser bug fixes.

[GitHub pull request](https://github.com/Cisco-Talos/clamav/pull/1225)
Fixed a bug causing some text to be truncated when converting from UTF-16.

[GitHub pull request](https://github.com/Cisco-Talos/clamav/pull/1232)
Fixed assorted complaints identified by Coverity static analysis.

[GitHub pull request](https://github.com/Cisco-Talos/clamav/pull/1237)
Fixed a bug causing CVDs downloaded by the DatabaseCustomURL Freshclam config option to be pruned and then re-downloaded with every update.

[GitHub pull request](https://github.com/Cisco-Talos/clamav/pull/1240)
Added the new 'valhalla' database name to the list of optional databases in preparation for future work.

[GitHub pull request](https://github.com/Cisco-Talos/clamav/pull/1240)
Silenced a warning "Unexpected early end-of-file" that occured when scanning some PNG files.

[GitHub pull request](https://github.com/Cisco-Talos/clamav/pull/1216)

**Pull request merged**